### PR TITLE
569 create xchain midgard module

### PR DESCRIPTION
**Pull request by Thorian1te**
New module, xchain-midgard
* Generates client using openapi-generate-cli
* exports interface that exposes api functions
* alpha version

### CODE REVIEW

File: packages/xchain-midgard/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2020 THORChain
```
Comment (veado) [inline packages/xchain-midgard/LICENSE:3]: 2022

**Review by veado — DISMISSED** [review]

File: packages/xchain-midgard/package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    "clean": "rimraf lib/**",
+    "build": "yarn clean",
+    "generate:types": "yarn clean:types:midgard && yarn generate:types:midgard",
+    "generate:types:midgard": "TS_POST_PROCESS_FILE=./node_modules/.bin/prettier openapi-generator-cli generate -i https://midgard.thorchain.info/v2/swagger.json -g typescript-axios -o ./src/generated/midgardApi --reserved-words-mappings in=in",
+    "clean:types:midgard": "rimraf ./src/generated/midgardApi"
+  },
+  "repository": "https://github.com/xchainjs/xchainjs-lib",
+  "author": "THORChain",
+  "license": "MIT",
+  "devDependencies": {
```
Comment (veado) [inline packages/xchain-midgard/package.json:16]: Add `rimraf` to `devDependencies`

File: packages/xchain-midgard/package.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "xchain-midgard",
+  "version": "0.1.0-alpha",
+  "description": "Midgard module that  auto-generates all the client code using openapi-generator-cli",
+  "main": "index.js",
```
Comment (veado) [inline packages/xchain-midgard/package.json:5]: There is no `index.js` (yet)

File: packages/xchain-midgard/package.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  "scripts": {
+    "clean": "rimraf lib/**",
+    "build": "yarn clean",
```
Comment (veado) [inline packages/xchain-midgard/package.json:8]: `build` is used to compile `TS` sources into `JS` and to provide a `lib` folder - all needed to publish package at npm - check other xchain-*  packages

Comment (Thorian1te) [reply packages/xchain-midgard/package.json:8]: Thanks! will modify the script and add a tsconfig-build.json

File: packages/xchain-midgard/package.json (hunk 3 above)
Comment (Thorian1te) [reply packages/xchain-midgard/package.json:5]: Fixed 👍

File: packages/xchain-midgard/package.json (hunk 2 above)
Comment (Thorian1te) [reply packages/xchain-midgard/package.json:19]: Fixed 👍

File: packages/xchain-midgard/LICENSE (hunk 1 above)
Comment (Thorian1te) [reply packages/xchain-midgard/LICENSE:3]: Fixed 👍

File: packages/xchain-midgard/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+```
+
+## Examples
+Request data from midgardApi
+
+```
+import { Midgard } from '@xchainjs/xchain-midgard'
+
+this.baseUrl = MIDGARD_API_URL;
+this.apiConfig = new Configuration({ basePath: this.baseUrl });
```
Comment (mikewyszinski) [inline packages/xchain-midgard/README.md:24]: example code needs slight fixing

Comment (Thorian1te) [reply packages/xchain-midgard/README.md:24]: okay

**Review by mikewyszinski — APPROVED** [review]

**Pull request merged**